### PR TITLE
spades: try to remove spades-bwa

### PR DESCRIPTION
**Pull request by alienzj**
## Involved packages

 - spades

## Involved issue

Close # https://github.com/BioArchLinux/Packages/commit/7f21e904bb65155d0e2d220f59c07d9970ab5049#commitcomment-143917186

## Details

- [x] Tested in the local machine (largest 16G RAM) and it is passed without any issue
- [ ] Provide New Package
- [X] Fix the Packages
  - [x] PKGBUILD
  - [ ] lilac.yaml
  - [ ] lilac.py
- [x] Would like to continue to work with us

## Additional Note
1. add `gcc` to `makedepends`
2. update cmake parameters refer to https://github.com/ablab/spades/blob/main/spades_compile.sh#L225

### CODE REVIEW

File: BioArchLinux/spades/PKGBUILD [hunk 1]
```diff
@@ -12,10 +12,11 @@ depends=(
   python
 )
 makedepends=(
-  bzip2
+  gcc
```
Comment (hubutui) [inline BioArchLinux/spades/PKGBUILD:15]: gcc is the default compiler, I don't think we need to add it.

**Comment by starsareintherose** [comment]
my question is that the spades-bwa is same with bwa, why not use ln -s to solve it?

File: BioArchLinux/spades/PKGBUILD (hunk 1 above)
Comment (alienzj) [reply BioArchLinux/spades/PKGBUILD:15]: I used `makepkg` to test building process, then I got some errors.
After installing `gcc`, it works. That's why I added gcc into PKGBUILD.

**Comment by alienzj** [comment]
Sorry for my misunderstanding.

From the CMakeLists.txt file https://github.com/ablab/spades/blob/main/ext/src/bwa/CMakeLists.txt#L33, it appears that SPAdes does not use the most recent version of BWA, but instead includes a method to build BWA as part of the SPAdes compilation process. This approach simplifies the installation for users, as they can compile SPAdes and have BWA built alongside it without needing to separately install BWA.

Arch Linux users, embracing the KISS principle, can opt for a simplified SPAdes installation process. If the latest BWA is compatible with SPAdes, we can remove the integrated BWA build from SPAdes and instead provide a symbolic link from spades-bwa to the latest BWA installation.

Let me try this week.
Thanks.

**Comment by alienzj** [comment]
Unfortunately,
SPAdes used: https://github.com/ablab/spades/blob/main/src/common/alignment/bwa_index.cpp#L377
BWA used: https://github.com/lh3/bwa/blob/master/bwamem.h

The API is not the same one.
It is difficult to use lh3's BWA directly at this moment.

**Comment by starsareintherose** [comment]
Okay, great, thanks for your testing

File: BioArchLinux/spades/PKGBUILD (hunk 1 above)
Comment (starsareintherose) [reply BioArchLinux/spades/PKGBUILD:15]: But have you tried bioarchlinux-x86_64-build?

Comment (pekkarr) [reply BioArchLinux/spades/PKGBUILD:15]: `gcc` is a dependency of `base-devel`, so it should not be in makedepends. From [Arch wiki](https://wiki.archlinux.org/title/PKGBUILD#makedepends):

> The package [`base-devel`](https://archlinux.org/packages/?name=base-devel) is assumed to be already installed when building with makepkg. Dependencies of this package **should not** be included in `makedepends` array.

Comment (alienzj) [reply BioArchLinux/spades/PKGBUILD]: Thanks. Removed `gcc`.

**Comment by starsareintherose** [comment]
you said not compatible, but why do you now use ln?

**Comment by alienzj** [comment]
I checked the API more in depth, basically, `mem_align1` did same thing when compared to `mem_align1_bin`.

**Comment by starsareintherose** [comment]
Temporarily closed because of no response.

**Pull request closed without merging**